simespresso/io_QE/espresso_class.py

```diff
@@ -0,0 +1,815 @@
+import logging
+import os.path
+import subprocess
+import sys
+
+import numpy as np
+from enum import Enum
+from simphony.core.cuba import CUBA
+from simphony.core.data_container import DataContainer
+from simphony.cuds.lattice import Lattice
+from simphony.cuds.particles import Particle, Particles
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class qe_functions(object):
+    '''
+    functions for reading and writing quantum espresso input and output files
+    '''
+
+    def __init__(self):
+        self.SP = DataContainer()  # System Model Equations and Material
+        # relations (Governing Equations)
+        self.SD = DataContainer()  # System Material Description and
+        # State Data including Boundaries (not conditions)
+        self.BC = DataContainer()  # Boundary conditions
+        self.CM = DataContainer()  # Computational Methods
+        # (numerical and solver aspects only)
+        self.pc = Particles('quantum_espresso_particles')
+
+    def read_espresso_output_file(self, file_name):
+        '''
+        This function parses  Espresso output files which usually will have
+        a name like 'name.charge'.
+        This file has the structure shown at
+        http://phycomp.technion.ac.il/~sbgrosso/QE_charge_density/node18.html
+        The first line has 8 numbers, the three first numbers are the size of
+        the grid in the three spatial directions, the three following are the
+        same repeated.
+        The seventh number corresponds to the number of atoms and the last
+        one to the number of different type of atoms.
+        Second line - the first number is the type of Bravais lattice,
+        and the three following numbers are the celldimensions defined in the
+        input file of pw.x.
+        The charge density is defined for each point of the grid starting
+        after the atom definitions.
+        :param file_name: name of the espresso output file
```

Comment:
we are using numpy-style documenation (see https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#documenting-classes).  

https://publicwiki-01.fraunhofer.de/SimPhoNy-Project/index.php/Development_guide#Programming_Languages_Specific_guidelines